Cannot connect after upgrade to 0.10.4

### xrdp version

0.10.4

### Detailed xrdp version, build options

I'm not running 0.10.4 anymore as it didn't work. Got through https://aur.archlinux.org/packages/xrdp
~~~
Paste the result between `~~~`.  Please DO NOT remove `~~~`!
~~~


### Operating system & version

Manjaro Linux

### Installation method

dnf / apt / zypper / pkg / etc

### Which backend do you use?

xorgxrdp

### What desktop environment do you use?

KDE

### Environment xrdp running on

Hyper-V

### What's your client?

Microsoft RDP client

### Area(s) with issue?

Crashes such as segfault

### Steps to reproduce

- Turn on enhanced mode in hyper-v
- Get xrdp login screen
- Enter login details
- Screen resizes to full screen
- Client disconnected

### ✔️ Expected Behavior

Expected to log into the desktop after screen resizing to fullscreen

### ❌ Actual Behavior

The client (microsoft rdp client) disconnects

### Anything else?

```
Jul 07 08:42:21 systemd[1]: Started xrdp daemon.
Jul 07 08:42:21 xrdp[909]: [WARN ] Config parameter 'use_vsock' is obsolete. Use 'port=vsock://...' instead
Jul 07 08:42:21 xrdp[909]: [INFO ] address [-1] port [3389] mode 3
Jul 07 08:42:21 xrdp[909]: [INFO ] listening to port 3389 on -1
Jul 07 08:42:21 xrdp[909]: [INFO ] xrdp_listen_pp done
Jul 07 08:42:21 xrdp[909]: [CORE ] You are running xrdp as root. This is not safe.
Jul 07 08:42:21 xrdp[909]: [INFO ] starting xrdp with pid 909
Jul 07 08:42:21 xrdp[909]: [INFO ] Socket 14: connection accepted from AF_VSOCK:cid=2/port=3949613062
Jul 07 08:42:21 xrdp[1200]: [INFO ] Using default X.509 certificate: /etc/xrdp/cert.pem
Jul 07 08:42:21 xrdp[1200]: [INFO ] Using default X.509 key file: /etc/xrdp/key.pem
Jul 07 08:42:21 xrdp[1200]: [ERROR] libxrdp_force_read: header read error
Jul 07 08:42:21 xrdp[1200]: [ERROR] [ITU-T X.224] Connection Sequence: CR-TPDU (Connection Request) failed
Jul 07 08:42:21 xrdp[1200]: [ERROR] xrdp_sec_incoming: xrdp_iso_incoming failed
Jul 07 08:42:21 xrdp[1200]: [ERROR] xrdp_rdp_incoming: xrdp_sec_incoming failed
Jul 07 08:42:21 xrdp[1200]: [ERROR] xrdp_process_main_loop: libxrdp_process_incoming failed
Jul 07 08:42:21 xrdp[1200]: [ERROR] xrdp_iso_send: trans_write_copy_s failed
Jul 07 08:42:21 xrdp[1200]: [ERROR] Sending [ITU T.125] DisconnectProviderUltimatum failed
Jul 07 08:45:42 xrdp[909]: [INFO ] Socket 14: connection accepted from AF_VSOCK:cid=2/port=3949613389
Jul 07 08:45:42 xrdp[4318]: [INFO ] Using default X.509 certificate: /etc/xrdp/cert.pem
Jul 07 08:45:42 xrdp[4318]: [INFO ] Using default X.509 key file: /etc/xrdp/key.pem
Jul 07 08:45:42 xrdp[4318]: [INFO ] Client requested security types (RDP assumed) : SSL|HYBRID|HYBRID_EX
Jul 07 08:45:42 xrdp[4318]: [INFO ] Selected classic RDP security
Jul 07 08:45:42 xrdp[4318]: [ERROR] libxrdp_force_read: header read error
Jul 07 08:45:42 xrdp[4318]: [ERROR] Processing [ITU-T T.125] Connect-Initial failed
Jul 07 08:45:42 xrdp[4318]: [ERROR] [MCS Connection Sequence] receive connection request failed
Jul 07 08:45:42 xrdp[4318]: [ERROR] xrdp_sec_incoming: xrdp_mcs_incoming failed
Jul 07 08:45:42 xrdp[4318]: [ERROR] xrdp_rdp_incoming: xrdp_sec_incoming failed
Jul 07 08:45:42 xrdp[4318]: [ERROR] xrdp_process_main_loop: libxrdp_process_incoming failed
Jul 07 08:45:42 xrdp[4318]: [ERROR] xrdp_iso_send: trans_write_copy_s failed
Jul 07 08:45:42 xrdp[4318]: [ERROR] Sending [ITU T.125] DisconnectProviderUltimatum failed
Jul 07 08:45:42 xrdp[909]: [INFO ] Socket 14: connection accepted from AF_VSOCK:cid=2/port=3949613390
Jul 07 08:45:42 xrdp[4320]: [INFO ] Using default X.509 certificate: /etc/xrdp/cert.pem
Jul 07 08:45:42 xrdp[4320]: [INFO ] Using default X.509 key file: /etc/xrdp/key.pem
Jul 07 08:45:42 xrdp[4320]: [INFO ] Client requested security types (RDP assumed) :
Jul 07 08:45:42 xrdp[4320]: [INFO ] Selected classic RDP security
Jul 07 08:45:42 xrdp[4320]: [INFO ] Connected client computer name: MY-HOSTNAME
Jul 07 08:45:42 xrdp[4320]: [INFO ] xrdp_load_keyboard_layout: Keyboard information sent by the RDP client, keyboard_type:[0x04], keyboard_subtype:[0x00], keylayout:[0x00000409]
Jul 07 08:45:42 xrdp[4320]: [INFO ] xrdp_load_keyboard_layout: model [] variant [] layout [us] options []
Jul 07 08:45:42 xrdp[4320]: [INFO ] client supports gfx protocol
Jul 07 08:45:42 xrdp[4320]: [WARN ] Physical desktop dimensions (0x0) are invalid
Jul 07 08:45:42 xrdp[4320]: [WARN ] Client and server do not both support the same encryption.
```

I looked through all logs in journalctl but can't find much, I've downgraded to 0.10.3 which works

Perhaps there is some config migration needed?

--- a/common/xrdp_client_info.h
+++ b/common/xrdp_client_info.h
@@ -172,7 +172,6 @@ struct xrdp_client_info
     int require_credentials; /* when true, credentials *must* be passed on cmd line */
 
     int security_layer; /* SECURITY_LAYER_* */
-    int vmconnect; /* Used when used from inside Hyper-V */
 
     int multimon; /* 0 = deny , 1 = allow */
     struct display_size_description display_sizes;
@@ -254,7 +253,7 @@ struct xrdp_client_info
     // Can we resize the desktop by using a Deactivation-Reactivation Sequence?
     enum client_resize_mode client_resize_mode;
 
-    int pad1; /* unused; unicode_input_state */
+    int vmconnect; /* Used when used from inside Hyper-V */
 };
 
 enum xrdp_encoder_flags

